src/test/java/com/buyanywhere/productcatalog/dto/CategoryDtoTest.java

```diff
@@ -0,0 +1,100 @@
+package com.buyanywhere.productcatalog.dto;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class CategoryDtoTest {
+
+    private CategoryDto categoryDto;
+
+    @Before
+    public void setup(){
+        categoryDto = new CategoryDto();
+        categoryDto.setId((long) 1);
+        categoryDto.setName("Home");
+        categoryDto.setDisplayOrder(1);
+    }
+
+    @Test
+    public void getId_shouldReturnId(){
+        Assert.assertTrue(this.categoryDto.getId() == 1);
+    }
+
+    @Test
+    public void getName_shouldReturnName(){
+        Assert.assertEquals(this.categoryDto.getName(), "Home");
+    }
+
+    @Test
+    public void getDisplayOrder_shouldReturnDisplayOrder(){
+        Assert.assertEquals(this.categoryDto.getDisplayOrder(), 1);
+    }
+
+    @Test
+    public void isValid_whenNameIsNull_shouldReturnFalse(){
+        CategoryDto dto = new CategoryDto();
+
+        Assert.assertFalse(dto.isValid());
+    }
+
+    @Test
+    public void isValid_whenNameIsEmpty_shouldReturnFalse(){
+        CategoryDto dto = new CategoryDto();
+        dto.setName("");
+
+        Assert.assertFalse(dto.isValid());
+    }
+
+    @Test
+    public void isValid_whenNameIsValidAndDisplayOrderIsNotValid_shouldReturnFalse(){
+        CategoryDto dto = new CategoryDto();
+        dto.setName("DisplayOrderIsNotValid");
+        dto.setDisplayOrder(-1);
+
+        Assert.assertFalse(dto.isValid());
+    }
+
+    @Test
+    public void isValid_whenNameAndDisplayOrderAreValid_shouldReturnFalse(){
```

Comment:
Copy Pasting pattern is not allowed unless you use the CCP (Carefully Copy & Paste Pattern) pattern instead. The name of the test should reflect what is expected to test. Instead of isValid_whenNameAndDisplayOrderAreValid_shouldReturnFalse, the name should be isValid_whenNameAndDisplayOrderAreValid_shouldReturnTrue